client/src/pages/board/components/dashboardForms/dialogSections/components/TextFieldOnFocusTypography.js

```diff
@@ -0,0 +1,101 @@
+import React, {useState, useCallback} from 'react';
+import {makeStyles, TextField, Typography, Box} from '@material-ui/core';
+
+// TODO: Documentation
+// TODO: Consider UX implications (e.g. no
+//       undo once component is blurred)
+
+/**
+ * @description
+ * A Component to Switch between a typography
+ * and a text field component.
+ *
+ */
+
+const useStyles = makeStyles((theme) => ({
+  inputMultiline: {
+    lineHeight: '1.1em',
+    padding: theme.spacing(2),
+  },
+  typography: {
+    padding: theme.spacing(2),
+    lineHeight: '1.1em',
+    borderRadius: theme.shape.borderRadius,
+    minHeight: theme.typography.body1.lineHeight * 4,
+    backgroundColor: ({disabled}) =>
+      disabled ? theme.palette.grey[200] : theme.palette.grey[300],
+    '&:hover': {
+      backgroundColor: ({disabled}) =>
+        disabled ? null : theme.palette.grey[400],
+      cursor: ({disabled}) => (disabled ? null : 'pointer'),
+    },
+  },
+}));
+
+const TextFieldOnFocusTypography = ({
+  TextFieldProps,
+  TypographyProps,
+  text,
+  saveText,
+  placeholder,
+  disabled,
+}) => {
```

Comment:
Really like how you set Inputs <-> Typography into its own component. I might try to modularize and reuse this, because its useful for any future instances of this rather than just CardDetails.

Reply:
Thanks, and yeah, I believe this would be a useful general purpose component!